src/sentry/integrations/github/integration.py

```diff
@@ -145,13 +145,14 @@ def reinstall(self) -> None:
     def message_from_error(self, exc: Exception) -> str:
         # TODO(mgaeta): Clean up the conditional flow.
         if isinstance(exc, ApiError):
-            message = API_ERRORS.get(exc.code, "")
-            if exc.code == 404 and re.search(r"/repos/.*/(compare|commits)", exc.url):
+            code = exc.code or -1
```

Comment:
The -1 feels a bit weird here, is this just because mypy complained about types if we attempted to use `None` as a key?

Reply:
Yeah I got:
```
error: Argument 1 to "get" of "Mapping" has incompatible type "Optional[int]"; expected "int"
```
I think I can fix this it up by adding more conditionals.